### PR TITLE
Exif writing fixes: Rational boundaries and signed/unsigned types

### LINKED ISSUE #3973: exif_transpose errors out on some images
### What did you do?
Used new `exif_transpose` method.

### What did you expect to happen?
I expected a transposed image to be returned, without error.

### What actually happened?
An error was raised.
```
/Users/okor/thumbor/lib/python2.7/site-packages/PIL/TiffImagePlugin.py:612: UserWarning: Metadata Warning, tag 282 had too many entries: 2, expected 1
  tag, len(values)))
/Users/okor/thumbor/lib/python2.7/site-packages/PIL/TiffImagePlugin.py:612: UserWarning: Metadata Warning, tag 283 had too many entries: 2, expected 1
  tag, len(values)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/okor/thumbor/lib/python2.7/site-packages/PIL/ImageOps.py", line 549, in exif_transpose
    transposed_image.info["exif"] = exif.tobytes()
  File "/Users/okor/thumbor/lib/python2.7/site-packages/PIL/Image.py", line 3087, in tobytes
    return b"Exif\x00\x00"+head+ifd.tobytes(offset)
  File "/Users/okor/thumbor/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 818, in tobytes
    count = len(data)
TypeError: object of type 'int' has no len()
```

Note that this error is not raised for _all_ images. Only some. I'm not sure how the EXIF data differs.

### What are your OS, Python and Pillow versions?

* OS: macOS Mojave 10.14.5
* Python: 2.7.16
* Pillow: 6.0.0

```python
from PIL import Image, ImageOps

im = Image.open('10_years_of_Wikipedia_by_Guillaume_Paumier.jpg')
im = ImageOps.exif_transpose(im)
```
[10_years_of_Wikipedia_by_Guillaume_Paumier.jpg.zip](https://github.com/python-pillow/Pillow/files/3398388/10_years_of_Wikipedia_by_Guillaume_Paumier.jpg.zip)

---

This error was encountered when attempting to replace `piexif` exif functionality with new Pillow functionality. The PR and conversation is here https://github.com/thumbor/thumbor/pull/1209

### LINKED ISSUE #4037: EXIF tobytes: ushort format requires 0 <= number <= (0x7fff * 2 + 1)
1. Download this image: https://cdn.everypony.ru/storage/06/18/58/2019/08/19/3e8e6e42ca.jpg

2. Run `Image.open('3e8e6e42ca.jpg').getexif().tobytes()`

3. Not sure if this is a corrupt image or a Pillow bug, but:

```
PIL/TiffImagePlugin.py:605: UserWarning: Metadata Warning, tag 283 had too many entries: 2, expected 1
  % (tag, len(values))

  File "PIL/Image.py", line 3213, in tobytes
    return b"Exif\x00\x00" + head + ifd.tobytes(offset)
  File "PIL/TiffImagePlugin.py", line 824, in tobytes
    data = self._write_dispatch[typ](self, *values)
  File "PIL/TiffImagePlugin.py", line 659, in <lambda>
    b"".join(self._pack(fmt, value) for value in values)
  File "PIL/TiffImagePlugin.py", line 659, in <genexpr>
    b"".join(self._pack(fmt, value) for value in values)
  File "PIL/TiffImagePlugin.py", line 626, in _pack
    return struct.pack(self._endian + fmt, *values)
struct.error: ushort format requires 0 <= number <= (0x7fff * 2 + 1)
```

P. S. Why Tiff?

### PR DESCRIPTION
**Pull request by kkopachev**
Related: #3973.

Changes proposed in this pull request:

 * Fix RATIONAL and SRATIONAL boundaries when writing IFDs
 * Chose signed/unsigned types based on passed value for unknown tags (fixes #4037)

### CODE REVIEW

**Comment by kkopachev** [comment]
I am totally dropping the ball at this one. 
I am not really able to come up with something working and elegant in terms of writing IFDs/sub-IFDs.
So let's merge this as is, at least some improvement which can be incorporated in next release. 

This seems to fix #4037. I didn't add very specific test case for it; it's kind of covered by same condition for RATIONAL/STRATIONAL tests. Let me know if I should add a specific test.

**Comment by kkopachev** [comment]
Let me know, if you prefer this rebased/squashed.

File: Tests/test_file_tiff_metadata.py [hunk 1]
```diff
@@ -136,14 +136,6 @@ def test_write_metadata(self):
...
         reloaded = loaded.tag_v2.named()
 
-        for k, v in original.items():
-            if isinstance(v, IFDRational):
-                original[k] = IFDRational(*_limit_rational(v, 2 ** 31))
-            elif isinstance(v, tuple) and isinstance(v[0], IFDRational):
-                original[k] = tuple(
-                    IFDRational(*_limit_rational(elt, 2 ** 31)) for elt in v
-                )
-
```
Comment (radarhere) [inline Tests/test_file_tiff_metadata.py]: Why remove this part of the test?

Comment (kkopachev) [reply Tests/test_file_tiff_metadata.py]: It seems disfunctional and wrong. 
The fact that Pillow used to incorrectly limit unsigned rationals to `2**31`, does not need to be validated. TIFF allows for rationals to be up to 2**32. If original file contained bigger (but still valid) value, then it has to be the same when Pillow saves it.

**Pull request merged**